x/perpetual/keeper/pool_health.go

```diff
@@ -85,3 +88,37 @@ func (k Keeper) CheckMinimumCustodyAmt(ctx sdk.Context, poolId uint64) error {
 	}
 	return nil
 }
+
+func (k Keeper) GetPoolTotalBaseCurrencyLiabilities(ctx sdk.Context, pool types.Pool) (sdk.Coin, error) {
+	// retrieve base currency denom
+	entry, found := k.assetProfileKeeper.GetEntry(ctx, ptypes.BaseCurrency)
+	if !found {
+		return sdk.Coin{}, errorsmod.Wrapf(assetprofiletypes.ErrAssetProfileNotFound, "asset %s not found", ptypes.BaseCurrency)
+	}
+	baseCurrency := entry.Denom
+
+	totalLiabilities := math.LegacyZeroDec()
+	for _, poolAsset := range pool.PoolAssetsLong {
+		// for long, liabilities will always be in base currency
+		totalLiabilities = totalLiabilities.Add(poolAsset.Liabilities.ToLegacyDec())
+	}
+
+	tradingAsset := ""
+	for _, poolAsset := range pool.PoolAssetsLong {
+		if poolAsset.AssetDenom != baseCurrency {
+			tradingAsset = poolAsset.AssetDenom
```

Comment:
@avkr003 you can break the loop right after assigning the value to `tradingAsset` as there is no need to continue iterating over the loop afterwards